### PR TITLE
Fix: Correct relationship display on workspace forms

### PR DESCRIPTION
**Pull request by amosmachora**
## Requirements

- [x] This PR has a title that briefly describes the work done, including the ticket number if there is a ticket.
- [x] My work conforms to the [**OpenMRS 3.0 Styleguide**](https://om.rs/styleguide).
- [x] I checked for feature overlap with [**existing widgets**](https://om.rs/directory).

## Summary
In this PR I have refactored the **family-relationship other-relationship & case-management workspaces ** to use the correct display property on the UI while maintaining the integrity of the relationships themselves. Previously the when creating a relationship for example a **sibling** we were using the display property which caused the UI to have something like **SIBLING/SIBLING**. This PR correctly identifies the relationship direction of the subjects and finds out the correct display value to use improving the UX of the app.

Also I correctly handled the cyclic dependency for the relationships. For example right now if you create a relationship of `Parent -> Child` That child now has a parent . I guess that functionality was there but now the UI shows the correct information.   

### Before
image

### After
image

### Other Relationships Workspace (After)
image

## Related Issue

*None.*

## Other

*None.*

### CODE REVIEW

**Comment by ojwanganto** [comment]
Thanks, @amosmachora. I defer to @donaldkibet and @makombe for their reviews.

File: packages/esm-patient-clinical-view-app/src/case-management/workspace/case-management.resource.ts [hunk 1]
```diff
@@ -50,11 +52,31 @@ export const useCaseManagers = () => {
...
+  const customRepresentation = 'custom:(uuid,display,displayAIsToB,displayBIsToA)&q=Case manager';
   const url = `/ws/rest/v1/relationshiptype?v=${customRepresentation}`;
   const { data, error } = useSWRImmutable<{ data: RelationshipTypeResponse }>(url, openmrsFetch);
 
-  return { data, error };
+  const tmp: {
+    uuid: string;
+    display: string;
+    direction: string;
+  }[] = [];
```
Comment (donaldkibet) [inline packages/esm-patient-clinical-view-app/src/case-management/workspace/case-management.resource.ts:63]:
```suggestion
 const relations: Array<{ uuid: string; display: string; direction: string }> = [];
```
Always good to have descriptive names for variables

File: packages/esm-patient-clinical-view-app/src/family-partner-history/relationships.resource.tsx [hunk 2]
```diff
@@ -79,13 +86,75 @@ function mapObservations(obsData) {
...
 export const useAllRelationshipTypes = () => {
-  const customRepresentation = 'custom:(uuid,display)';
-  const url = `${restBaseUrl}/relationshiptype?v=${customRepresentation}`;
+  const url = `${restBaseUrl}/relationshiptype?v=default`;
   const { data, error } = useSWRImmutable<{ data: RelationshipTypeResponse }>(url, openmrsFetch);
 
   return { data, error };
 };
 
+export const useFamilyRelationshipTypes = () => {
```
Comment (donaldkibet) [inline packages/esm-patient-clinical-view-app/src/family-partner-history/relationships.resource.tsx:95]: These two hooks look identical to me why don't we have one hook e.g `useRelationShipType` that returns both type i.e `familyRelationship` and `defaultRelationship`

File: packages/esm-patient-clinical-view-app/src/family-partner-history/relationships.resource.tsx [hunk 3]
```diff
@@ -79,13 +86,75 @@ function mapObservations(obsData) {
...
+};
+
+// Hook for getting all other relationship types other than "family types";
+export const useOtherRelationshipTypes = () => {
+  const url = `${restBaseUrl}/relationshiptype?v=default`;
+  const { data, error } = useSWRImmutable<{ data?: RelationshipTypeResponse }>(url, openmrsFetch);
+  const { familyRelationshipsTypeList } = useConfig<ConfigObject>();
+  const familyRelationshipTypesUUIDs = new Set(familyRelationshipsTypeList.map((r) => r.uuid));
+
+  const tmp: RelationshipType[] = [];
```
Comment (donaldkibet) [inline packages/esm-patient-clinical-view-app/src/family-partner-history/relationships.resource.tsx:133]:
```suggestion
  const relationShipType: Array<RelationshipType> = [];
```

**Review by donaldkibet — DISMISSED** [review]
LGTM Left some suggestions Thanks @amosmachora 🤝

**Comment by amosmachora** [comment]
@donaldkibet are we good now. I refactored the hooks to avoid code duplication.

File: packages/esm-patient-clinical-view-app/src/case-management/workspace/case-management.resource.ts [hunk 4]
```diff
@@ -50,11 +52,27 @@ export const useCaseManagers = () => {
...
+  const customRepresentation = 'custom:(uuid,display,displayAIsToB,displayBIsToA)&q=Case manager';
   const url = `/ws/rest/v1/relationshiptype?v=${customRepresentation}`;
   const { data, error } = useSWRImmutable<{ data: RelationshipTypeResponse }>(url, openmrsFetch);
 
-  return { data, error };
+ const relations: Array<{ uuid: string; display: string; direction: string }> = [];
+
+  data?.data.results.forEach((type) => {
+    const aIsToB = {
+      display: type.displayAIsToB ? type.displayAIsToB : type.displayBIsToA,
```
[the remark this answers is not in the document]
Comment (amosmachora) [reply packages/esm-patient-clinical-view-app/src/case-management/workspace/case-management.resource.ts:63]: That is how they have it implemented in the community. So I figured they probably had a good reason why.

[the remark this answers is not in the document]
Comment (amosmachora) [reply packages/esm-patient-clinical-view-app/src/case-management/workspace/case-management.resource.ts:63]: Do you think I should remove it ?

**Comment by ojwanganto** [comment]
Please share the latest screenshot/video

**Comment by amosmachora** [comment]
@ojwanganto here is the video you requested.

https://github.com/user-attachments/assets/f166a079-0197-4f3e-a127-a0b264484842

**Review by ojwanganto — DISMISSED** [review]
Thanks, @amosmachora

File: packages/esm-patient-clinical-view-app/src/case-management/workspace/case-management.resource.ts [hunk 5]
```diff
@@ -49,12 +51,30 @@ export const useCaseManagers = () => {
   return { data, error };
 };
 
-export const useRelationshipType = () => {
-  const customRepresentation = 'custom:(uuid,display)&q=Case manager';
+export const useCaseManagerRelationshipType = () => {
```
Comment (ojwanganto) [inline packages/esm-patient-clinical-view-app/src/case-management/workspace/case-management.resource.ts:54]: is this hook just for case management relationship types? I think you should revert to the previous variable name.

Comment (ojwanganto) [inline packages/esm-patient-clinical-view-app/src/case-management/workspace/case-management.resource.ts:54]: I think we can use the useRelationships hook and just filter the case manager relationships. What do you think?

**Comment by amosmachora** [comment]
Here is a screenshot to confirm the changes made have not broken anything.

image

and on case management.

image

**Comment by ojwanganto** [comment]
Thanks, @amosmachora. This looks good

**Comment by ojwanganto** [comment]
@donaldkibet please check this one more time so that we have it in

**Review by ojwanganto — APPROVED** [review]

**Pull request merged**